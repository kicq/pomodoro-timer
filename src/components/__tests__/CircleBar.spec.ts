import { mount } from '@vue/test-utils'
import CircleBarVue from '../CircleBar.vue'


describe('CircleBar.vue', () => {

  const props = {
    size: 50,
    thin: 10
  }

  const wrapper = mount(CircleBarVue, {
    props
  })

  it('should be correct structure', () => {

    const svg = wrapper.element.querySelectorAll('svg')[0]
    const circles = svg.querySelectorAll('circle')

    expect(svg).toBeTruthy()
    expect(circles.length).toBe(2)
    expect(wrapper.element.children.item(1)?.tagName === 'DIV').toBeTruthy()
  })

  it('check svg radius', () => {
    const [svg] = wrapper.element.querySelectorAll('svg')
    const [circle, circle2] = svg.querySelectorAll('circle')
    const calcResult = String(props.size / 2 - props.thin / 2)
    expect(circle.getAttribute('r')).toBe(calcResult)
    expect(circle2.getAttribute('r')).toBe(calcResult)
  })

  it('check snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})