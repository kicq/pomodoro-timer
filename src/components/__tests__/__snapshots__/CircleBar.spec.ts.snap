// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleBar.vue check snapshot 1`] = `
<div class="circle-bar"><svg id="svg" width="50" height="50" viewPort="0 0 25 25" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <circle r="20" cx="25" cy="25" fill="transparent" stroke-width="10px" stroke-dasharray="157.07963267948966" stroke-dashoffset="0"></circle>
    <circle id="bar" r="20" cx="25" cy="25" fill="transparent" stroke-width="10px" stroke-dasharray="157.07963267948966" stroke-dashoffset="117.80972450961724px"></circle>
  </svg>
  <div class="legend" style="font-size: 16px;">02:05</div>
</div>
`;
